OpenEdXMobile/src/main/java/org/edx/mobile/view/Router.java

```diff
@@ -508,4 +511,50 @@ public void showFeedbackScreen(@NonNull FragmentActivity activity, @NonNull Stri
                 .append(activity.getString(R.string.insert_feedback));
         EmailUtil.openEmailClient(activity, to, subject, body.toString(), config);
     }
+
+    public void showFeedbackScreen(
+            @NonNull FragmentActivity activity,
+            @NonNull String subject,
+            @Nullable Integer errorCode,
+            @Nullable Integer errorEndpoint,
+            @Nullable String errorMessage) {
+        final String NEW_LINE = "\n";
+        final String to = config.getFeedbackEmailAddress();
+        StringBuilder body = new StringBuilder();
+        body.append(String.format("%s %s", activity.getString(R.string.android_os_version), android.os.Build.VERSION.RELEASE))
+                .append(NEW_LINE)
+                .append(String.format("%s %s", activity.getString(R.string.app_version), BuildConfig.VERSION_NAME))
+                .append(NEW_LINE)
+                .append(String.format("%s %s", activity.getString(R.string.android_device_model), Build.MODEL))
+                .append(NEW_LINE)
+                .append(getFormattedErrorMessage(activity, errorCode, errorEndpoint, errorMessage))
+                .append(NEW_LINE)
+                .append(activity.getString(R.string.insert_feedback));
+        EmailUtil.openEmailClient(activity, to, subject, body.toString(), config);
+    }
+
+    private StringBuilder getFormattedErrorMessage(
```

Comment:
better to move this method in `org.edx.mobile.util.TextUtils` OR in some other appropriate utils class.